timemachine/cpp/src/kernels/k_nonbonded_precomputed.cuh

```diff
@@ -15,167 +15,170 @@ void __global__ k_nonbonded_precomputed(
                     energy += FLOAT_TO_FIXED_ENERGY<RealType>(nrg);
                 }
-            }
-
-            if (du_dx || du_dp) {
-                RealType d_ij_12 = d_ij_6 * d_ij_6;
-                RealType du_dr = eps_ij * 24 * sig_ij_6 * (d_ij_6 - 2 * sig_ij_6) / (d_ij_12 * d_ij);
-
-                RealType force_prefactor = du_dr * inv_dij;
-                if (du_dx) {
-                    gi_x += FLOAT_TO_FIXED_NONBONDED(delta_x * force_prefactor);
-                    gi_y += FLOAT_TO_FIXED_NONBONDED(delta_y * force_prefactor);
-                    gi_z += FLOAT_TO_FIXED_NONBONDED(delta_z * force_prefactor);
-
-                    gj_x += FLOAT_TO_FIXED_NONBONDED(-delta_x * force_prefactor);
-                    gj_y += FLOAT_TO_FIXED_NONBONDED(-delta_y * force_prefactor);
-                    gj_z += FLOAT_TO_FIXED_NONBONDED(-delta_z * force_prefactor);
 
-                    atomicAdd(du_dx + atom_i_idx * 3 + 0, gi_x);
-                    atomicAdd(du_dx + atom_i_idx * 3 + 1, gi_y);
-                    atomicAdd(du_dx + atom_i_idx * 3 + 2, gi_z);
-
-                    atomicAdd(du_dx + atom_j_idx * 3 + 0, gj_x);
-                    atomicAdd(du_dx + atom_j_idx * 3 + 1, gj_y);
-                    atomicAdd(du_dx + atom_j_idx * 3 + 2, gj_z);
+                if (du_dx || du_dp) {
+                    RealType du_dr = q_ij * es_factor;
+
+                    RealType force_prefactor = du_dr * inv_dij;
+                    if (du_dx) {
+                        // forces
+                        gi_x += FLOAT_TO_FIXED_NONBONDED(delta_x * force_prefactor);
+                        gi_y += FLOAT_TO_FIXED_NONBONDED(delta_y * force_prefactor);
+                        gi_z += FLOAT_TO_FIXED_NONBONDED(delta_z * force_prefactor);
+
+                        gj_x += FLOAT_TO_FIXED_NONBONDED(-delta_x * force_prefactor);
+                        gj_y += FLOAT_TO_FIXED_NONBONDED(-delta_y * force_prefactor);
+                        gj_z += FLOAT_TO_FIXED_NONBONDED(-delta_z * force_prefactor);
+                    }
+
+                    if (du_dp) {
+                        // du/dp
+                        g_q_ij += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DCHARGE>(erfc_beta * inv_dij);
+                        g_dw_ij += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DW>(delta_w * force_prefactor);
+                    }
                 }
+            }
 
-                if (du_dp) {
-                    RealType du_ds = -24 * eps_ij * pow(sig_ij, 5) * (d_ij_6 - 2 * sig_ij_6) / d_ij_12;
-
-                    g_eps_ij += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DEPS>(du_de);
-                    g_sig_ij += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DSIG>(du_ds);
-                    g_dw_ij += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DW>(delta_w * force_prefactor);
+            if (eps_ij != 0 && sig_ij != 0) {
+                RealType d_ij_4 = d2_ij * d2_ij;
```

Comment:
nit: preexisting, but consider naming consistency (`d2_ij` vs. `d_ij_4`)

Reply:
Fixed to match `compute_lj` (https://github.com/proteneer/timemachine/blob/26d693dc3ac562377b8e5300f2c019e0e95c63b3/timemachine/cpp/src/kernels/k_nonbonded_common.cuh#L69) in https://github.com/proteneer/timemachine/pull/1300/commits/cf2bb8d0bf326b645b86bcf75f77bac7888f26f5